lib/call-cabal-project-to-nix.nix

```diff
@@ -527,23 +562,31 @@ let
     export SSL_CERT_FILE=${cacert}/etc/ssl/certs/ca-bundle.crt
     export GIT_SSL_CAINFO=${cacert}/etc/ssl/certs/ca-bundle.crt
 
-    # Using `cabal v2-freeze` will configure the project (since
-    # it is not configured yet), taking the existing `cabal.project.freeze`
-    # file into account.  Then it "writes out a freeze file which
-    # records all of the versions and flags that are picked" (from cabal docs).
+    export CABAL_DIR=$(mktemp -d)
+
+    cat >$CABAL_DIR/config <<EOF
+    repository hackage.haskell.org
+      url: http://hackage.haskell.org/
+      secure: True
+      root-keys: aaa
+      key-threshold: 0
```

Comment:
Apparently if I don't put this `root-keys`, I get a validation failure, despite the presence of `key-threshold: 0`. :shrug: